modules/geolocationRtdProvider.js

```diff
@@ -0,0 +1,64 @@
+import {submodule} from '../src/hook.js';
+import {isFn, logError, pick, cleanObj} from '../src/utils.js';
+import {config} from '../src/config.js';
+import {getGlobal} from '../src/prebidGlobal.js';
+import '../src/adapterManager.js';
+
+let permissionsAvailable = true;
+let geolocation;
+function getGeolocationData(requestBidsObject, onDone, providerConfig, userConsent) {
+  let done = false;
+  if (!permissionsAvailable) return complete();
+  const requestPermission = providerConfig.requestPermission === true;
+  const waitForIt = providerConfig.waitForIt;
+  const auctionDelay = config.getConfig('realTimeData.auctionDelay') || 0;
+  const adUnits = requestBidsObject.adUnits || getGlobal().adUnits || [];
+  navigator.permissions.query({
+    name: 'geolocation',
+  }).then(permission => {
+    if (permission.state !== 'granted' && !requestPermission) return complete();
+    navigator.geolocation.getCurrentPosition(geo => {
+      geolocation = pick(geo.coords, [
+        'accuracy',
+        'altitude',
+        'altitudeAccuracy',
+        'heading',
+        'latitude',
+        'longitude',
+        'speed'
+      ]);
+      complete();
+    });
+  });
+  if (auctionDelay > 0) {
+    setTimeout(complete, auctionDelay);
+  }
+  if (!waitForIt) complete();
+  function complete() {
+    if (done) return;
+    done = true;
+    geolocation && adUnits && adUnits.forEach((unit) => {
+      unit.bids && unit.bids.forEach(bid => bid.geolocation = cleanObj(geolocation));
```

Comment:
Instead of tacking this on to bid requests, it's IMO better to fill this out as FPD (setting the fields in `requestBidsObject.ortb2Fragments.global.device.geo`). That way a number of bidders will automatically forward that data, and [activity controls](https://github.com/prebid/Prebid.js/pull/9802) will automatically control the accuracy. 

Otherwise this relies entirely on more work in bid adapters so that they may pick this up. Looking at the [spec](https://github.com/InteractiveAdvertisingBureau/openrtb2.x/blob/main/2.6.md#objectgeo) it looks like most of the fields have a spot with the exception of those about velocity, I'd still put those somewhere under `geo.ext`.

Reply:
i'd prefer the browser doesn't even ask for precise geo unless https://github.com/InteractiveAdvertisingBureau/Global-Privacy-Platform/blob/87935c4b574e974ff20ff5c1b9641374878031e2/Sections/US-National/IAB%20Privacy%E2%80%99s%20National%20Privacy%20Technical%20Specification.md?plain=1#L229 permitted it; not geo gets asked for then rounded without the proper consent. Does activityControls allow for that? we'd want to prevent invoking this altogether, not just manipulating its input to bid requests.

Reply:
@patmmccann just to be clear, i need to check consent or just ensure that consent string provided ?

Reply:
@AdmixerTech after discussing this offline, this should _not_ use the browser API if the `transmitPreciseGeo` activity is denied. Activities are very new, but the basic idea is to do something like

```javascript
if (isActivityAllowed(ACTIVITY_TRANSMIT_PRECISE_GEO, activityParams(MODULE_TYPE_RTD, 'geolocation'))) {
   // call browser geo api
}
```

Some examples of the activity API in use for reference:

https://github.com/prebid/Prebid.js/blob/bdd56a742c07df9dd9a2025e93b1649ce187d0bb/src/storageManager.js#L32-L35

https://github.com/prebid/Prebid.js/blob/bdd56a742c07df9dd9a2025e93b1649ce187d0bb/src/adapterManager.js#L281-L285